### PR TITLE
Added guide on profiling and improving performance

### PR DESCRIPTION
**Pull request by Darylgolden**
## Overview: What does this pull request change?
I added a guide explaining how contributors may help profile the library, along with specific instructions for cProfile and SnakeViz.

## Links to added or changed documentation pages

https://manimce--2517.org.readthedocs.build/en/2517/contributing/performance.html

## Reviewer Checklist
- [ ] The PR title is descriptive enough for the changelog, and the PR is labeled correctly
- [ ] If applicable: newly added non-private functions and classes have a docstring including a short summary and a PARAMETERS section
- [ ] If applicable: newly added functions and classes are tested

### CODE REVIEW

File: docs/source/contributing/performance.rst [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+Now, we can run SnakeViz on the profile file:
+
+.. code-block:: bash
+
+   snakeviz square_to_circle.txt
+
+A browser window or tab will open with a visualization of the profile, which should
+look something like this:
+
+.. image:: _static/snakeviz.png
```
Comment (behackl) [inline docs/source/contributing/performance.rst:80]: Image is not loaded correctly. One of these two fixes it (I think), not sure whether the first one works.

```suggestion
.. image:: /_static/snakeviz.png
```

```suggestion
.. image:: ../_static/snakeviz.png
```

**Review by behackl — CHANGES_REQUESTED** [review]
Image is not loaded, but other than that LGTM. Thanks for putting this together!

**Comment by behackl** [comment]
Draft?

**Comment by Darylgolden** [comment]
> Draft?

Yeah, a code block isn't rendering for some reason.

File: docs/source/contributing/performance.rst [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+   pip install snakeviz
+
+cProfile is included with in Python's standard library and does not need to be installed.
+
+Suppose we want to profile ``SquareToCircle``. Then we add and save the following code
+to ``square_to_circle.py``:
+
+.. code-block:: python
+    from manim import *
```
Comment (behackl) [inline docs/source/contributing/performance.rst:48]: (Some?) Directives require a newline between them and their content.

```suggestion

    from manim import *
```

**Review by behackl — APPROVED** [review]
LGTM now!

**Pull request merged**